src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,76 @@
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
+            throw new IllegalArgumentException("Interface class must be annotated with @Component");
+        }
+        Object clazzImplementationInstanse = null;
+        Class<?> clazz = findImlementation(interfaceClazz);
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstanse = createNewInstance(clazz);
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstanse, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("cant initialize field " + field.getName()
+                            + "class: " + clazz.getName());
+                }
+            }
+
+        }
+        if (clazzImplementationInstanse == null) {
+            clazzImplementationInstanse = createNewInstance(clazz);
+        }
+        return clazzImplementationInstanse;
+    }
+
+    private Class<?> findImlementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+        if (interfaceClazz.isInterface()) {
+            return interfaceImplementations.get(interfaceClazz);
+        }
+        return interfaceClazz;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException | SecurityException | InstantiationException
+                 | IllegalAccessException | IllegalArgumentException
+                 | InvocationTargetException e) {
+            throw new RuntimeException("cant instantiate " + clazz.getName());
```

Comment:
Replace the multiple exceptions in the catch block with a general parental exception 'ReflectiveOperationException' to simplify the exception handling. Also, include the cause of the exception and a more descriptive error message.